src/github.com/getlantern/flashlight/config/config.go

```diff
@@ -122,7 +119,8 @@ func Init(userConfig UserConfig, version string, configDir string, stickyConfig
 		m.ValidateConfig = validateConfig
 		m.DefaultConfig = MakeInitialConfig
 		m.CustomPoll = func(ycfg yamlconf.Config) (mutate func(yamlconf.Config) error, waitTime time.Duration, err error) {
-			return fetcher.pollForConfig(ycfg)
+			// Request the config via either chained servers or direct fronted servers.
+			return pollForConfig(ycfg, newFetcher(userConfig, proxied.ParallelPreferChained(), flags, "global.yaml.gz"))
```

Comment:
It looks like we can use the same logic to fetch/serve/persist global config and proxies. What you did in `proxies.go` is much more cleaner to me than the callbacks in `yamlconf`.